ext/parquet/parquet_test.go

```diff
@@ -22,6 +24,76 @@ type mockStruct struct {
 	Timestamp  time.Time `parquet:"timestamp_col"`
 }
 
+func (m *mockStruct) UnmarshalParquet(obj interfaces.UnmarshalObject) error {
```

Comment:
instead of that, is there a way to declare a `Timestamp CustomTimeWrapper` on the struct? is there maybe a custom type available withing `goparquet`?

Reply:
looks like https://pkg.go.dev/github.com/fraugster/parquet-go/floor#Time is what you need to use

Reply:
sorry, looks like floor supports scanning into time.Time - https://github.com/fraugster/parquet-go/blob/v0.3.0/floor/reader.go#L230 but it requires logical type to be set correctly to DATE or TIMESTAMP, is this the case on our test AND "production" files? can you check them manually and see what is being set?

Reply:
https://github.com/fraugster/parquet-go/issues/35

Reply:
Yes. `floor` is supporting scanning into time when the schema defines the column as one of 

```
message test {
	required int32 date (DATE);
	required int64 tsnano (TIMESTAMP(NANOS, true));
	required int64 tsmicro (TIMESTAMP(MICROS, true));
	required int64 tsmilli (TIMESTAMP(MILLIS, true));
	required int64 tnano (TIME(NANOS, true));
	required int64 tmicro (TIME(MICROS, true));
	required int32 tmilli (TIME(MILLIS, true));
}
```
Even if the logical type was set I doubt it would decode. `floor.Scan` doesn't seem to support int96. So if the logical type was set it would try to get the int value here https://github.com/fraugster/parquet-go/blob/c28e3072a9721f25d59193b9d74c642d20800fe8/floor/reader.go#L178 and would fail.